traitlets/traitlets.py

```diff
@@ -489,11 +488,24 @@ def _validate(self, obj, value):
         return value
 
     def _cross_validate(self, obj, value):
-        if hasattr(obj, '_%s_validate' % self.name):
-            cross_validate = getattr(obj, '_%s_validate' % self.name)
-            value = cross_validate(value, self)
+        try:
+            cb = getattr(obj, '_%s_validate' % self.name)
+            if cb in obj._trait_validators.get(self.name,[]):
+                raise
```

Comment:
I think the problem with that is people will probably just want to slap a decorator on existing functions and change the signature in order to use the new api. Defaulting to the old signature in the case of a conflict doesn't seem right.